src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -668,6 +741,156 @@ InferenceEngine::Precision FullyConnected::getRuntimePrecision() const {
     return getMaxPrecision(inputPrecisions);
 }
 
+void FullyConnected::initOptimalPrimitiveDescriptor() {
+    Node::initOptimalPrimitiveDescriptor();
+    auto selectedPD = getSelectedPrimitiveDescriptor();
+    implementationTypeIP = selectedPD->getImplementationType();
+    // if convolution selected the reorder for ip is useless. Delay the reoder for ip to prepareParams
+    const auto constParent = getParentEdgeAt(1)->getParent();
+    const auto selectedParentPD = constParent->getSelectedPrimitiveDescriptor();
+    auto config = selectedPD->getConfig();
+    weightDescIP = config.inConfs[1].getMemDesc();
+    config.inConfs[1].setMemDesc(selectedParentPD->getConfig().outConfs[0].getMemDesc());
+    selectedPD->setConfig(config);
+}
+
+DnnlDesriptor FullyConnected::createDescriptorInternalForConv(const dnnl::memory::desc &inputDesc,
+                                const dnnl::memory::desc &biasDesc,
+                                const dnnl::memory::desc &outputDesc) const {
+    dnnl::memory::data_type wdt = inputDesc.data_type();
+    dnnl::memory::data_type bdt = outputDesc.data_type();
+    if (inputDesc.data_type() == dnnl::memory::data_type::bf16) {
+        bdt = dnnl::memory::data_type::f32;
+    } else if (inputDesc.data_type() == dnnl::memory::data_type::u8 || inputDesc.data_type() == dnnl::memory::data_type::s8) {
+        wdt = memory::data_type::s8;
+        if (withBiases)
+            bdt = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalInputPrecisionAtPort(BIAS_ID));
+    }
+
+    // make a fake shape: N, IC, 1, W
```

Comment:
Why do we need 4th rank, I though 3d conv are available as well.

Reply:
Yes, it's ok when rank equals 3,4,5, and I think rank=4 which means 2d conv is more general and has best testcase coverage, so prefer rank=4.

Reply:
In the [code](https://github.com/openvinotoolkit/oneDNN/blob/7a0d09627701d16550033a13b114f201f2312fb2/src/cpu/x64/jit_brgemm_conv_utils.cpp#L1497) it will make the spatial dimension to 1d when kernel's stride statisfied the condition. Since this may make worry I changed it to rank = 3.